web/pdf_find_controller.js

```diff
@@ -142,59 +142,36 @@ var PDFFindController = (function PDFFindControllerClosure() {
     },
     calcSearchMatch: function PDFFindController_calcSearchMatch(pageIndex) {
       var pageContent = this.normalize(this.pageContents[pageIndex]);
-      var query = this.normalize(this.state.query);
-      var caseSensitive = this.state.caseSensitive;
-      var queryLen = query.length;
+      var matches = [];
+      var queryLens= [];
 
-      if (queryLen === 0) {
-        return; // Do nothing: the matches should be wiped out already.
-      }
+      //Nirdosh: regular expression to match full sentence
+      for(var i=0;i<this.state.query.terms.length;i++){
+        var matchIdx = -queryLen;
+        var query = this.normalize(this.state.query.terms[i]);
+        var caseSensitive = this.state.caseSensitive;
+        var queryLen = query.length;
+        if (queryLen === 0) {
+          continue; // Do nothing: the matches should be wiped out already.
+        }
 
-      if (!caseSensitive) {
+        if (!caseSensitive) {
         pageContent = pageContent.toLowerCase();
         query = query.toLowerCase();
-      }
-
-      var matches = [];
-      var queryLens= [];
-      var matchIdx = -queryLen;
-      /*while (true) {
-        matchIdx = pageContent.indexOf(query, matchIdx + queryLen);
-        if (matchIdx === -1) {
-          break;
         }
-        matches.push(matchIdx);
-      }*/
-      //Nirdosh: regular expression to match full sentence
-      var regex=new RegExp('[^.]* '+query+' [^.]*\.','gi');
-      var match;
-      while ((match = regex.exec(pageContent)) !== null) {
-        if (match.index === regex.lastIndex) {
-             ++regex.lastIndex;
+
+        //Nirdosh: regular expression to match full sentence
+        var regex=new RegExp("[^.]* "+query+"[^.]*\.","gi");
+        var match;
+        // console.log(query);
+        while ((match = regex.exec(pageContent)) != null) {
```

Comment:
Compare with !== to compare value and type:
http://www.w3schools.com/js/js_comparisons.asp